sigstore/verify/verifier.py

```diff
@@ -143,6 +143,7 @@ def staging(cls) -> Verifier:
 
     def verify(
         self,
+        input_: IO[bytes],
         materials: VerificationMaterials,
         policy: VerificationPolicy,
     ) -> VerificationResult:
```

Comment:
@jku @laurentsimon do either of you have strong opinions about the input type here? I made it `IO[bytes]` to mirror our current `sign(...)` API, but (IMO) using IO types has been a weakness for us (it pushes effects into the API here, makes the API stateful, etc.).

As an alternative, I'm thinking of making this just take a `bytes` instead -- I think that'll be perfectly fine for ordinary input sizes, and users with huge inputs can use the `Hashed` interface (once I add it here).

Reply:
I've changed this to take `bytes` instead, per thoughts above. Once this is merged, I'll do a similar refactor PR for `sign(...)`.